cataclysm-bn-git: init at 551c7279dafe1d283c3ef2779cb335b855e6bc59

caryoscelus opened the pull request:
Add Cataclysm: Bright Nights, a fork/variant of Cataclysm: Dark Days Ahead. This adds git version, stable to be added later

As per discussion in #330693 , this is implemented as "sub-package" of C:DDA because of virtually the same build procedure. A notable difference is introduction of desktopFilePath because C:BN seems to lack one

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

caryoscelus committed:
cataclysm-bn-git: init at 551c7279dafe1d283c3ef2779cb335b855e6bc59

Add Cataclysm: Bright Nights, a fork/variant of Cataclysm: Dark Days Ahead
This adds git version, stable to be added later

caryoscelus commented:
pinging people who showed interest from previous PR: @DeeUnderscore @cizra @rardiol

cizra commented:
Works fine, both the tiles and console versions.

DeeUnderscore commented on pkgs/games/cataclysm-dda/bn-git.nix:
```diff
@@ -0,0 +1,38 @@
+      "VERSION=git-${version}-${lib.substring 0 8 src.rev}"
+    ];
+
+    meta = common.meta // {
+      maintainers = with lib.maintainers;
+      common.meta.maintainers ++ [ rardiol ];
+    };
```
I would recommend adding `description` and `longDescription` that says this is a fork of Cataclysm: Dark Days Ahead, mentioning the full title (Cataclysm: Bright Nights) in the `longDescription`. People might find this via package search and be confused otherwise.

FliegendeWurst commented on pkgs/games/cataclysm-dda/bn-git.nix:
```diff
@@ -0,0 +1,38 @@
+  });
+in
+
+attachPkgs pkgs self
```
You need to use `desktopFilePath` here, otherwise it does not eval.

caryoscelus replied:
not sure what you mean.. it does work for me and replacing with `attachPkgs pkgs self desktopFilePath` makes build break

FliegendeWurst replied:
I don't remember either

caryoscelus committed:
Merge remote-tracking branch 'upstream/master' into cataclysm-bn

caryoscelus committed:
cataclysm-bn-git: 2024-10-18 -> 2025-05-14

Add sqlite as dependency & patch out dangling symlinks

caryoscelus committed:
cataclysm-bn-git: fix meta

caryoscelus commented:
@DeeUnderscore added, also fixed other meta entries, thanks!

sorry for such a long pause, I've updated PR to current version (had to add a patch that doesn't feel entirely good; proper way would be to either replace symlink with a copy or install doc/ in package, but it works fine for now)

caryoscelus committed:
Fix merge typo

FliegendeWurst commented on pkgs/games/cataclysm-dda/bn-git.nix:
```diff
@@ -0,0 +1,63 @@
+        monsters.
+      '';
+      homepage = "https://cataclysmbn.org/";
+      maintainers = with lib.maintainers; common.meta.maintainers ++ [ caryoscelus ];
```
You need to add yourself to `maintainers/maintainer-list.nix` :)

caryoscelus replied:
done

FliegendeWurst commented on pkgs/games/cataclysm-dda/bn-git.nix:
```diff
@@ -0,0 +1,63 @@
+
+  self = common.overrideAttrs (common: rec {
+    pname = "cataclysm-bn-git";
+    inherit version;
```
usual naming scheme would be "0-unstable-YYYY-MM-DD", so `version = "0-unstable-${version}";` here

FliegendeWurst replied:
Or rather, `(last stable version)-unstable-...`

caryoscelus replied:
fixed

caryoscelus committed:
cataclysm-bn-git: fixes

- add @caryoscelus to maintainer list
- fix version schema
- bump to latest git

DeeUnderscore commented on pkgs/games/cataclysm-dda/bn-git.nix:
```diff
@@ -0,0 +1,63 @@
+      inherit rev sha256;
+    };
+
+    patches = [
+      # Unfortunately some READMEs in data are symlinks to doc which is not included
+      # resulting in dangling symlinks breaking build
+      ./delete-dangling-readme.patch
+    ];
```
It would probably be simpler to get rid of the patch file, and just do something like

```suggestion
    postInstall = ''
      # Unfortunately some READMEs in data are symlinks to doc which is not included
      # resulting in dangling symlinks breaking build
      rm $out/share/cataclysm-bn/json/{external_tileset/README.md,mapgen/lab/README.md}
    '';
```

(I'm not sure if this is the exact path, but this is the idea)

DeeUnderscore reviewed (changes requested):
You will need to run the new file through the formatter to satisfy the checks (sorry, the neighboring files were not formatted yet when you opened the PR, so they were not a good example of the current standard). See the [formatting section of CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#formatting) for how to.

Also, please squash down the commits to ideally just two:

1. maintainers: add caryoscelus
2. cataclysm-bn-git: init at 0.8.0-unstable-2025-05-17

in this order
